lib/iiif/fedora2.rb

```diff
@@ -60,15 +80,14 @@ def pages
           ]
         else
           # look up the METS relations in Fedora 2 to get a list of image PIDs
-          xml_doc = Nokogiri::XML(get_mets_xml)
-          fptrs = xml_doc.xpath(
+          fptrs = mets.xpath(
             '/mets:mets/mets:structMap[@TYPE="LOGICAL"]/mets:div[@ID="images"]//mets:div[@ID="DISPLAY"]/mets:fptr',
             mets: METS_NAMESPACE
           )
           fptrs.map do |fptr|
-            label = fptr.xpath('../..').attribute('LABEL').value
+            label = doc ? doc['displayLabel'] : fptr.xpath('../..').attribute('LABEL').value
```

Comment:
@cfitz This needs to check not only if `doc` is defined, but also if the `displayLabel` is in fact present. We are losing the labels for the canvases otherwise (they are all being rendered as `null`).